rust/agama-cli/src/config.rs

```diff
@@ -108,67 +107,67 @@ pub async fn run(subcommand: ConfigCommands, opts: GlobalOpts) -> anyhow::Result
-            let store = SettingsStore::new(http_client.clone()).await?;
+            let http_client = build_http_client(api_url, opts.insecure, true).await?;
             let url_or_path = url_or_path.unwrap_or(CliInput::Stdin);
             let contents = url_or_path.read_to_string(opts.insecure)?;
             let valid = validate(&http_client, CliInput::Full(contents.clone()), false).await?;
 
-            if matches!(valid, ValidationOutcome::Valid) {
-                let result =
-                    InstallSettings::from_json(&contents, &InstallationContext::from_env()?)?;
-                tokio::spawn(async move {
-                    show_progress(monitor, true).await;
-                });
-                store.store(&result).await?;
+            if !matches!(valid, ValidationOutcome::Valid) {
+                return Ok(());
             }
 
-            Ok(())
+            let model: api::Config = serde_json::from_str(&contents)?;
+            patch_config(&http_client, &model).await?;
         }
         ConfigCommands::Validate { url_or_path, local } => {
             let _ = if !local {
-                let (http_client, _monitor) = build_clients(api_url, opts.insecure).await?;
+                let http_client = build_http_client(api_url, opts.insecure, true).await?;
                 validate(&http_client, url_or_path, false).await
             } else {
                 validate_local(url_or_path, opts.insecure)
             };
-
-            Ok(())
         }
         ConfigCommands::Generate { url_or_path } => {
-            let (http_client, _monitor) = build_clients(api_url, opts.insecure).await?;
+            let http_client = build_http_client(api_url, opts.insecure, true).await?;
             let url_or_path = url_or_path.unwrap_or(CliInput::Stdin);
 
-            generate(&http_client, url_or_path, opts.insecure).await
+            generate(&http_client, url_or_path, opts.insecure).await?;
         }
         ConfigCommands::Edit { editor } => {
-            let (http_client, monitor) = build_clients(api_url, opts.insecure).await?;
-            let store = SettingsStore::new(http_client.clone()).await?;
-            let model = store.load().await?;
+            let http_client = build_http_client(api_url, opts.insecure, true).await?;
+            let response: api::Config = http_client.get("/v2/config").await?;
             let editor = editor
                 .or_else(|| std::env::var("EDITOR").ok())
                 .unwrap_or(DEFAULT_EDITOR.to_string());
-            let result = edit(&http_client, &model, &editor).await?;
-            tokio::spawn(async move {
-                show_progress(monitor, true).await;
-            });
-            store.store(&result).await?;
-            Ok(())
+            let result = edit(&http_client, &response, &editor).await?;
+            patch_config(&http_client, &result).await?;
         }
     }
+
+    Ok(())
+}
+
+async fn patch_config(
```

Comment:
Alternatively, we could have a single client which implements this logic. It could be a wrapper of the `BaseClient` or the `BaseClient` itself.